packages/devextreme/js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1214,18 +1214,24 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
     },
 
     _renderSelectAllItem: function($container) {
+        const { selectAllText, focusStateEnabled } = this.option();
         $container = $container || this.$element().find(`.${NODE_CONTAINER_CLASS}`).first();
 
         this._$selectAllItem = $('<div>').addClass(SELECT_ALL_ITEM_CLASS);
 
         const value = this._dataAdapter.isAllSelected();
-        this._createComponent(this._$selectAllItem, CheckBox, {
+        const selectAllCheckBox = this._createComponent(this._$selectAllItem, CheckBox, {
             value: value,
             elementAttr: { 'aria-label': 'Select All' },
-            text: this.option('selectAllText'),
+            text: selectAllText,
+            focusStateEnabled,
             onValueChanged: this._onSelectAllCheckboxValueChanged.bind(this)
         });
 
+        selectAllCheckBox.registerKeyHandler('enter', () => {
```

Comment:
I think we may declare registerKeyHandler using onInitialized callback as follows: 
```
...
onInitialized: ({ component }) => {
    component.registerKeyHandler('enter', () => {
        component.option('value', !component.option('value'));
    });
}
...
```

Reply:
your right, updated